### ISSUE TITLE
Add emoji reactions on updates

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#4185]
## User story 
_From https://opencollective.slack.com/archives/G46PNRCGP/p1617912913040900_
> updates needs an emoji function? I just went in high five  the latest open web docs update, but I can only comment. I don't really want to say anything, but a heart would be nice.

## Specs
This feature is really the same as comment reactions, the only difference is that these ones would be indexed on `UpdateId` instead of `CommentId`. We'll re-use `CommentReactions` to achieve that, renaming it to `EmojiReactions` to make it generic.

### API

- [x] Rename `CommentReactions` table & model to `EmojiReactions`
- [x] Add an `UpdateId` column in there (nullable)
- [x] Make `CommentId` nullable
- [x] Update the `['CommentId', 'FromCollectiveId', 'emoji']` index to index where `UpdateId=NULL` (conditional index)
- [x] Add a new conditional index on `['UpdateId', 'FromCollectiveId', 'emoji']` (where `CommentId=NULL`)
- [x] Rename existing `addReaction` method to `addReactionOnComment`
- [x] Add a new `addReactionOnUpdate` method
- [x] Deprecate `addUpdateReaction`
- [x] Create a new `addEmojiReaction` mutation that works seamlessly on updates or comments
- [x] Do the same for `removeCommentReaction`
- [x] Add a `reactionsByUpdateId` loader next to `reactionsByCommentId`
- [x] On `Update` GraphQL object, add a `reactions` (JSON) field
- [x] Add matching unit tests

### Frontend

- [x] Allow users to add/remove reactions (use the same layout than comments)
- [x] Show reactions below the update

**Comment by SudharakaP** [comment]
@Betree : I've added the support for this. There's some small stuff to do I think which I'll update as I go along. 

- Tests needs to be added; I'll do this today or tomorrow. 😄 

- I think we need to refactor the code in the front end; for example the class names I think should be changed to remove the dependency on comments as now the emojis are generic to both comments and updates. So maybe we rename things like, `CommentReactions.js` to `Reactions.js` etc. Let me know if you agree with this. I am thinking of doing this in a separate PR. 😄 

- I've deprecated the `CommentReactionMutations` in this PR; although the initial post says `Deprecate addUpdateReaction`. I suppose this is what you meant right? There's no `addUpdateReaction` in the API. 😉 

Anyways, let me know if you see any issues with the PRs. Thanks much. 😄 🙏🏽

**Comment by Betree** [comment]
> So maybe we rename things like, CommentReactions.js to Reactions.js etc. Let me know if you agree with this. I am thinking of doing this in a separate PR. smile

Definitely, though I would rename it to `EmojiReactions` to keep it explicit.

**Comment by SudharakaP** [comment]
@Betree : I've changed my mind to rename the files in the same PR as it's not difficult, hope it's okay. I've added the tests and also made some refactoring both on front and api to make sure we rename the file names and get rid of references to strings such as `CommentReactions` to make it more generic with `EmojiReactions`. Take a look and let me know if you see any issues. 😄 

Also I've opened a new issue to track the deprecation of `CommentReactionMutations.js`; https://github.com/opencollective/opencollective/issues/4393
